### ISSUE TITLE
[UX] 3-dot toolbar menus should all be consistent across browser vs homescreen (toolbar menu redesign)

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#14320]
## User Story
- As a user, I want to see the same options in the same order (when applicable) in the toolbar so it is easy for me to get used to them and be able to quickly interact with them.

### Acceptance Criteria
- I see the same toolbar options when I am on different screens (homescreen, tab, etc)
- I find it easy to find what I am looking for regardless of what screen I am on (measured through user testing)

**Comment by liuche** [comment]
Also see #14394 for addons menu item.

**Comment by liuche** [comment]
cc @violasong bc I think you're working on this this sprint!

**Comment by violasong** [comment]
Issues:
- **Improve the ordering of Android menu items**
- Consider trade-off of reachability with thumb vs. ease of reading top-to-bottom
- Consider Desktop Firefox's menu ordering
- Feedback: Some people look at the top of the menu for important actions
- Feedback: 'Bookmark' and 'Bookmarks' are easily missed or hard to find
- Which options would also be appropriate for the 3 dot menu on tabs tray? (Settings?)
- Concern: The bottom toolbar menu being a completely reversed version about the top could be confusing
  - Should that menu be the same as the top toolbar menu? 
  - What about a hybrid approach: Each section stays in a logical order (e.g. Bookmarks should always be at the top of its section), but the sections could be reversed?

Next steps: 
- Get data on most used options from release
- Closer look at previous work on Save Panel and Share Panel and usability study
- Make mockups
- Investigate research options from Data Science
- Launch pilot user test

**Comment by HUYNHHOANGNAM** [comment]
Chương trình này là do tôi lập ra kế hoạch cho fixfox nhưng hiện tại tôi đang gặp khó khăn về tài chính không thể thực hiện được nếu được sự hỗ trợ tài chính tôi sẽ đưa fix fox trở lại với người dùng ưa chuộng email liên lạc của tôi huynhhoangnam2629@gmail.com
huynhnam0111994@gmail.com hoặc số điện thoại +840933832629

**Comment by TheRealOsas** [comment]
"Consider trade-off of reachability with thumb vs. ease of reading top-to-bottom" please can u guys keep this as an option, one thing I like about fenix is that everything can be reached at thumbs length (if your toolbar Is at the bottom)

**Comment by yoasif** [comment]
@violasong https://github.com/mozilla-mobile/fenix/issues/9871 is worth a look here.

**Comment by axeltrinidade** [comment]
Wouldn't the different honescreen and tab toolbar look more consistent if the background color of the additional tab toolbar options be slightly lighter/darker than the basic options (which appear in the honescreen)

**Comment by klint** [comment]
Hi. I was just wondering: is the "remote debugging by USB" needed where it is now, given that's a very advanced setting that is unlikely to be used by a regular user? Maybe in the "about Firefox" section near "Support" entry? Or permanently behind the "Secret Settings"? :)

**Comment by cadeyrn** [comment]
"About Firefox" or the "Secret Settings" are totally wrong for this setting. In fact it's a very important setting for a very large target audience: millions of people develop websites, either professional or as hobby. And to be honest I doubt that really _every_ other options in the settings has more users.

**Comment by klint** [comment]
> "About Firefox" or the "Secret Settings" are totally wrong for this setting. In fact it's a very important setting for a very large target audience: millions of people develop websites, either professional or as hobby. And to be honest I doubt that really _every_ other option in the settings has more users and if these don't get removed from the main settings screen…
Ok, got it, thanks for the explanation

> But I can't see how this issue is about settings. It's about the menues, no?
yes, it seems to be only about the level 1 menu actually. My bad.

So here is one thing for the level 1 menu: regarding Install vs Add to homescreen.
The code is ready to have both entries displayed at the same time instead of only one of them. yet only one is displayed currently, per design decision. Just asking whether having both displayed could be considered in this ticket ? (as a reminder, the use case is for instance when a news site like lemonde.fr is available as a PWA but when you would like to add a specific article to the homepage. This is not possible currently, as you can only install the site as a PWA that leads to the homepage, not to the article itself).

**Comment by violasong** [comment]
Related issue: #10344 involves changes to the tab tray menu. @topotropic is investigating a "Tab Settings" menu item which leads to a new page in Settings.

**Comment by violasong** [comment]
Update: Working on a survey based on the following mockup, and requesting meta-feedback from Firefox mobile UX, [twitter](https://twitter.com/Firefox_Android/status/1304184042703654912), [reddit](https://www.reddit.com/r/firefox/comments/iqebl7/hi_im_a_ux_designer_for_firefox_for_android/).

<img width="896" alt="toolbarmenusurvey@2x-8" src="https://user-images.githubusercontent.com/55994/92820737-24adfd80-f37f-11ea-86d0-56ed2acdb160.png">

**Comment by amkcpu** [comment]
Will the synced tabs entry be removed once  #14117 lands?
For now, it's needed, but afterwards, a separate menu entry seems redundant to me.

**Comment by klint** [comment]
Where will the "Open in application" and "Exit" menu entries be placed ? These entries are displayed or hidden depending on the context, but where will they be displayed when activated?

**Comment by amkcpu** [comment]
> Where will the "Open in application" and "Exit" menu entries be placed ?

Same question for the `Appearance` option that appears when you open a site in reader mode.
Or will this menu be moved into `Settings` > `Customize`?

**Comment by jawz101** [comment]
@violasong I think I like 3.  For some reason it feels like page actions are at the top, settings stuff in the middle, bookmarks and other organization/mgmt stuff at the bottom.  I guess it must be familiar already due to the current ui.

**Comment by Poopooracoocoo** [comment]
just wanna be able to drag to open and select a menu item

**Comment by gtg947h** [comment]
Speaking for myself, the problem is that menu items appear (or disappear) depending on what's open.  I know some UI genius says that UI elements should just be hidden entirely if you can't use them, but the problem is that menu items move up and down on the menu depending on whether I'm on the home page or an actual site.  This is a problem for very frequently sed hings because I get used to where they are 90% of the time and use them on muscle memory instead of reading, and end up selecting the wrong one the other 10% of the time because the one I want has moved.

Menu items should either gray out if unavailable, or the ones that never go away should be at the top (assuming a top menu bar) so they don't move around.

**Comment by JoGarWeb** [comment]
I will make some observations:
1. Why not hide some options that are usually used only once? (Hide them like the "addons" menu does).
2. Remember possible new features such as "Save in PDF" (Feature requested by users)
3. Important: Remember the "Quit" option.
4. I forgot the option "Install", in the same way it would be hidden in the menu. 

![Uploading 20200914_030436.jpg…]()

**Comment by Poopooracoocoo** [comment]
for an explanation of what I said, check out Flutter's issue. (sorry for the cross-link!) https://github.com/flutter/flutter/issues/5064 

People use this a lot in Chrome because it's even faster than native menus. This behaviour is also found in iOS/iPadOS and most desktop operating systems too.

**Comment by violasong** [comment]
Rough wireframe that shows a different approach: grid menus, similar to those of Edge & Samsung Internet.

Sounds like this could be a version 2, whereas the basic list reordering could be the MVP. 


<img width="572" alt="image" src="https://user-images.githubusercontent.com/55994/93385156-adfb7f00-f81a-11ea-91a1-7532acc9ca15.png">

**Comment by TheRealOsas** [comment]
Love this, I guess we have found the one.

**Comment by yoasif** [comment]
@violasong The latest image you just shared - do you picture it as a "tray"-like item or more of a [GNOME style popover](https://developer.gnome.org/hig/stable/popovers.html.en)/menu?

**Comment by klint** [comment]
Will there be a setting to remove the labels and keep the icon only in a minimal size menu?

**Comment by liuche** [comment]
Somewhat related: #14728 for changing the bookmark icon fill color to be more distinctive (Fennec has a bookmarked icon as blue).

**Comment by amkcpu** [comment]
Some questions for the new wireframe:
- Do the colors used for the first four icons have any meaning associated with them? I know they are used on iOS, but is there an original intention or a way for the user to make sense of them?
- Is there a reason for the synced tabs icon to be different from the currently used one?
- A more general issue with icons. There are different actions associated with them: Some are simple links (bookmarks, history), some point to an action (find in page, save to collection), some a toggle (desktop mode), and some sub-menus (add-ons). With the old menu, those intentions are pretty clear (e.g. desktop mode has a toggle on the right), but icons are the same for everything. How can the user be made aware of that? 
- A note on i18n: In some languages, the label names will be longer than in English and will be ellipsized. This is something to keep in mind, because it can influence the usability. Currently, the toolbar menu is a lot wider in some languages than in English to accommodate the longer labels. Can something similar be done for an icon grid? Maybe Multi-Line/2-line labels?

**Comment by cadeyrn** [comment]
@violasong 

> Rough wireframe that shows a different approach

If you're looking for a different approach you could also have a look at the menu of Firefox Lite:

![screenshot](https://user-images.githubusercontent.com/1100614/93662408-22732f80-fa60-11ea-914f-0c5bf01087a2.png)

It has the **big** advantage that you can reach really _every_ menu item with only one hand. I don't say it should be used 1:1 (and in fact it can't be used 1:1 because Firefox Lite has a very different feature set) but it could be an additional source for inspiration.

**Comment by klint** [comment]
Could the 1 or 2 mostly used share targets be added to the menu at page level? This was the case in Fennec if I remember correctly and was quite useful

**Comment by amkcpu** [comment]
![Above](https://github.com/mozilla-mobile/fenix/issues/14320#issuecomment-693965103), I commented that in the original mockup, the same type of icon is used to convey different actions. To address this, I drew up the following three mockups that try to separate the different actions.

## Intro
The idea is that the icons can be separated into three types:
- `Shortcuts`, that take you to specified areas in the app (e. g. downloads or history page)
- `Page actions`, that let you interact with the site (e. g. toggle desktop mode, find in page)
- `Save/share`, that let you take the site content elsewhere, e.g. saving it to the homescreen or sharing it in other apps.

Please note that the mockups are _very_ unfinished and not polished at all, it's more about a rough visualization. Also, the strings would probably need to be changed.

## Alternative 1
The mockup pulls page actions out of the icon grid and puts them in a separate row. Add-on pages are included in the page action row, and the row would be scrollable for overflowing content.
To address #14926, the share and send to device icons are moved to the "Share or save to" row.

<img src="https://user-images.githubusercontent.com/46764284/93670296-b9f57400-fa9a-11ea-8cde-ce4eebb1ea07.png" width=300 />

## Alternative 2
This further splits up the menu, moving share into a separate row. The advantage is that apps recently used for sharing can be shown (as it used to be in Fennec)

<img src="https://user-images.githubusercontent.com/46764284/93670302-c4b00900-fa9a-11ea-8f27-b5e82adce3a0.png" width=300 />

## Alternative 3
Because fundamentally, the upper menu shortcuts correspond to actions within the browser/app, and the lower ones correspond to site actions, one could add color to distinguish the two. IMO, color should be on the site-specific parts, because when the user is on the home screen, only the upper icon grid will be visible.
I used the same color as the one on the tab tray for the currently selected tab, but this could of course be a different color.

<img src="https://user-images.githubusercontent.com/46764284/93670195-d80ea480-fa99-11ea-9c7d-c2e1cd02c182.png" width=300 />


## Open questions/concerns
- Alternative 2 might be to high/large for small devices.
- Which icons should be round and with a background color, and which ones not filled?
- `Quit` is an action, not a shortcut, but I still placed it on the upper row because it belongs to the browser-wide actions. This might not be optimal.

**Comment by klint** [comment]
A setting to remove all strings (titles and legends) could be useful to get a compact menu, which would make sense once the user is used to all features.

**Comment by violasong** [comment]
Just launched a user test for the MVP menu sorting options. (As mentioned, this will be for the next sprint starting Thurs. The grid-style menu needs more time.)

<img width="1007" alt="image" src="https://user-images.githubusercontent.com/55994/93832626-21dac480-fc2b-11ea-88d2-c4f5d6498df4.png">

**Comment by violasong** [comment]
Thanks for everyone's comments so far. I'll respond about grid menu concepts after I get this first MVP completed.

**Comment by violasong** [comment]
Related issue: #11987

**Comment by rileyinman** [comment]
My preference on the grid design would also be number 2, as it would best facilitate a possible switch to the native share sheet (more details on what I mean in #14926).

**Comment by black7375** [comment]
@violasong 
How about using the design of Firefox Light? (@cadeyrn 's suggestion)
- Easy one-handed operation
- Neat design
- Code reusability

I have set a few criteria.

### Rule
- One-handed operation should be easy
- The lower the depth, the better
- Important features should always be accessible
- Must be well distinguished
- Satisfy other existing issues

### Group
- **Essential:** Forward, Backward, Refresh
- **Page Action:** Open in app, Find in page, Desktop site, Screenshot, Text size
- **Save:** Add to Bookmark, Add to top Site, Add to home Screen, Save to collection
- **Share:** Send to Device, Share to app(Recently App, More App)
- **Library:** Bookmark, History, Sync Tabs, Downloads
- **About:** Settings, Addons, Helps, Quit

### Component
- **List**
  - **Familiarity:** Portability of the current UX, Similar to desktop
  - **Easy to operate:** Only need to move your finger vertically
- **Icon**
  - **Stand out:** The icon is large
  - **Space Utilization:** Left and right swipe is possible
- **Icon(`Fixed`)**
  - **Accessibility:** Always accessible
  - **No Caption:** It always exists, Users remember their position.
  - **No Space:** It's the most accessible and should be used to the fullest.

### Order - Considerations
- **Frequency**
  - **High:** Fixed or visible.
  - **Nomal:** Relatively, the position can be changed or hidden.
- **Time**
  - **Immediately:** Must return to or reflect the page.
  - **Slowly:** Menu is deep or needs thinking like settings
- **Access Location**
  - **Inside:** Look forward to quick access.
  - **Outside:** Can stand it even if it is slow.

### Result Table
I made it assuming it's a long list.
|**Group**                       |**Component** |**Frequency**|**Time**    |**Access**|
|--------------------------------|--------------|-------------|------------|----------|
|**Library**                     | Icon         | High        | Slowly     | Inside   |
|**Page Action**                 | List         | Normal      | Immediately| Inside   |
|**Save**                        | List         | Normal      | Immediately| Inside   |
|**Share**                       | Icon         | Normal      | Slowly     | Outside  |
|**About**                       | List         | Normal      | Very Slowly| Inside   |
|**Essential**                   | Icon(`Fixed`)| High        | Immediately| Inside   |
- `About` is the last one, but it's not as slow as you think. (Fitts' Law)

### Result Sample
I am not a designer. It's a concept.
Grids and sizes can be weird...
![3-Dot_Menu_Origin](https://user-images.githubusercontent.com/25581533/94351227-1d0c7b00-0046-11eb-84c7-46201de5b3be.png)
-  I used text scaling as an example of easily setting a deeper setting. #14923 #15445

**Comment by JoGarWeb** [comment]
Again, I am here to ask you to hide the options that are usually not used. In the last updates of Firefox "Report Site Issue..." was moved to the main menu and now the menu is too big. 
Just a menu similar to the one with Add-ons for the options that are not often used for sites. I think users don't Report Site Issue, either add to the top sites, or make a shortcut on the homescreen to each site you visit. 

![](https://i.imgur.com/xp8EaIG.png)

**Comment by Poopooracoocoo** [comment]
tbh i wish addons weren't in that menu

**Comment by Dunexus** [comment]
Some addons have page-related settings, it makes sense to be able to reach them from the 3 dot menu as it has the other page-related options (add to bookmark, search in page, etc.).

**Comment by violasong** [comment]
Quick little update: I had to take a step back from this because of learning about new factors that need to be researched and analyzed. We're having a lot of discussions around balancing priorities right now. 

Thanks @black7375 for the mockup/proposal! I hope to have more feedback/info to share soon.

**Comment by violasong** [comment]
Hi all, I'm back! I've reviewed the comments here and looked at the latest data. A huge thanks to everyone who's weighed in with ideas. Special thanks to @cadeyrn and @black7375 for the Firefox Lite-based ideas. Here's my latest draft.

<img width="584" alt="image" src="https://user-images.githubusercontent.com/55994/104077671-8ec09d00-51cf-11eb-92bb-35ad59eb5cb7.png">

Next steps: 
1. Create a full-grid design and top toolbar versions
2. Follow up with Data Science and others re: questions I have about the data
3. Run user tests to determine usability. In particular, validate the following hypotheses:

- Moving Bookmarks into a horizontal library bar, next to its peers (History and Downloads), will improve the experience for those who confused it with the bookmark button. (See https://github.com/mozilla-mobile/fenix/issues/11987 - I need to replicate this milestone study from @apbitner.)
- Adding "New Tab" to this menu will help new users migrating from Chrome/Edge. (Consider New Private Tab as well)
- Even though Add-ons are currently not accessed very often compared to other items in this menu, Add-ons are an important differentiator for Firefox and benefit from a higher prominence in order to encourage discoverability.

**Comment by black7375** [comment]
It looks neat and much better than the current menu.
I also like the consideration for top toolbar users.

**Comment by klint** [comment]
Thank you for this very neat layout proposal.

I have 2 questions:
- Will this new layout allow to reopen #7758 (for PWA sites, have 2 entries in the menu instead of "Install" only: "Add to Homescreen" and "Install")? Visually speacking at least, it seems to be the case. And could be so useful (the background code is ready to display both entries, if I remember correctly. It is only a matter of decision).
- The "Login and passwords" screen is currently reachable with 4 clicks on Android, but only in 2 clicks on desktop, as it is in the main menu there. This is an option that may be used often to get a password to log in in the application instead of a webpage. Just asking whether it could be considered as a main menu option as well?

Thanks.

**Comment by dglttr** [comment]
Thanks a lot for your proposal!

I like it a lot, just two thoughts:
- `Settings` is not a menu item that is needed particularly often. Therefore, how about moving it down next to `Report Site Issue`? That would make the site actions more visible.
- Will the `Synced Tabs` entry be removed once https://github.com/mozilla-mobile/fenix/issues/14117 lands? It would be redundant then IMO.

**Comment by cadeyrn** [comment]
I agree with @dglttr regarding settings and synced tabs for the following reason:

This is the proposed order of menu items:

- New Tab => not page specific
- Desktop Site => page specific
- Find in Page => page specific
- Synced Tabs => not page specific
- Settings => not page specific
- Add to Collection => page specific
- Add to Top Sites => page specific
- Add to Homescreen => page specific
- Open in App => not page specific
- Report Site Issue => not page specific
- Quit => not page specific

So with the latest proposal page specific and non page specific actions are mixed. By moving the settings and removing the synced tabs items there are no longer non page specific actions between the page specific actions.

With the current menu it makes sense to have the synced tabs menu items because the other "library" pages like history and bookmarks are there as well. With the new proposal all of these items are moved to the top - except of the synced tabs. And once #14117 will land, it's no longer needed in the main menu anyway. And then the settings menu item will remain the only non page specific action between all of these page specific actions. After moving this to the bottom area there is with the "New Tab" menu item still a non page specific action at top but I guess it's okay because it could be considered as important enough to show it at first.

**Comment by violasong** [comment]
Thanks all!

@klint Re: Install and Logins & Passwords

I'm not sure about these, and would have to do more investigation - unfortunately, to keep this task on track, I probably won't be changing what we currently have as part of this project.

@dglttr

> Settings is not a menu item that is needed particularly often. Therefore, how about moving it down next to Report Site Issue? That would make the site actions more visible.

Settings is actually accessed surprisingly often (much more often than the site actions) according to our data.

> Will the Synced Tabs entry be removed once #14117 lands? 

Yes! This is good point. We'll just have to see when #14117 can land.

@cadeyrn 

Thanks for this analysis. I know they are a mixed which is unfortunate. Hopefully, pending progress on #14117 we can remove Synced from this menu. As for Settings, it's trickier because it's a highly accessed item. (We need to do a separate investigation into whether anything from Settings needs to be moved in a different location.) 

I think what could help is grouping the Desktop toggle and Settings together, since Desktop is a setting. So it would look like this:


<img width="615" alt="image" src="https://user-images.githubusercontent.com/55994/104244327-0a605b00-5417-11eb-9e57-04db2c296c4e.png">

**Comment by violasong** [comment]
Two more parts to this:

## Top toolbar mode
To keep the most important actions near the user's current activity, we would move the main actions bar to the top. This gives us some iffy alignment issues between the elements in the two top bars, so would need to do something like a background color to better visually separate the two.

<img width="268" alt="image" src="https://user-images.githubusercontent.com/55994/104258421-61c0f400-5434-11eb-82ef-e61e0a1443f4.png">

## Full grid menu 

I'm considering a full grid menu as an alternative. One benefit of this menu is that it's more compact, and doesn't need a truncated mode. We can also use a "Add to" heading for the Collection/Top Sites/Homescreen trio. I'm keeping that at the bottom for this layout because I don't want to chunk up the center part of the menu.

Downsides: Somewhat riskier UI choice, given that this isn't common. With all icons having the same prominence, it feels like a bit of icon overload — we would need to add color or other styling. Could look a little lopsided when the optional items aren't visible. I'm not sure if this is easier to scan than the list/grid hybrid. I'm planning to include this in user testing, but it's hard to say if I'll get firm enough signals from that.

<img width="274" alt="image" src="https://user-images.githubusercontent.com/55994/104258542-97fe7380-5434-11eb-8b1e-6c9089611e8a.png">

Version with colored library, inspired by Edge:

<img width="582" alt="image" src="https://user-images.githubusercontent.com/55994/104259980-620ebe80-5437-11eb-87d4-12cc4eea9fc9.png">

**Comment by CharoSW** [comment]
As you're talking about moving important actions to near the current activity would it make sense to just reverse the order?  With the bottom toolbar approach you have the main actions at bottom and the bookmarks etc. area capping the menu at the top, so if you're moving the main actions to the top would it not be better to have that bookmark etc. area capping the menu along the bottom?

**Comment by violasong** [comment]
New versions based on rounds of critique.

- Option 1: Removes bookmark button from main action bar, puts it into an "Save to" row with a label
- Option 2: Keeps bookmark button in main action bar, and also puts it in a "Save to" submenu for those who don't instantly recognize the star in the main menu

Feedback welcome!

<img width="1124" alt="Screen Shot 2021-01-19 at 12 13 39 PM" src="https://user-images.githubusercontent.com/55994/105087744-cd681a00-5a4f-11eb-9482-f54b83c1579e.png">

**Comment by dglttr** [comment]
Hi Victoria,
thanks for sharing your progress, it's really appreciated!

I personally prefer the right screen because it is visually cleaner and more structured:
- The upper three menu items focus on page-specific actions, whereas the screen on the right has settings mixed in.
- I also like that "Save to" is not taking up too much space, because most of these actions are not needed often.
- The bookmark icon is directly accessible _and_ serves as an indicator whether I have already bookmarked this page.

What do you think about exchanging "New tab" with "Report issue"? The idea is that "Report issue" is page-specific (thus belonging to "find in page" and "desktop site"), whereas "New tab" is not and is better placed in the "General menu items" section alongside "Settings".

**Comment by klint** [comment]
Thanks Victoria
I would prefer the second option for the reason explained by @dglttr
Yet, the new tab button seems too far from the thumb and would be better at the bottom, IMHO. Maybe this is another reason for exchanging it with the Report issue.
And last, where would the Install option appear? 'save to' what, on that case?

**Comment by cadeyrn** [comment]
I fully agree with @dglttr.

Additional notes: option 2 looks less cluttered. And the submenu for "Save to" has the advantage to make it easier to integrate more save options.

Example: In #15450 it was requested to always show a "Add to home screen" option even if the website is available as PWA. This design would make it possible while it would be more difficult with option 1.

Sure, you could move the bookmark icon to the last icon row (as in option 2) then you can integrate another option in the "Save to" row but option 2 still gives more flexibility if you want to integrate even more things in the future like… Save to Pocket? It would be the ideal place for task continuation apps.

Another consideration are translations. In your mockup of option 1 there is a "Home screen" item and it already needs two lines of text although it is not a very long term. But even this won't work with the German language. In Germany it's "Startbildschirm", no space between the characters so you can't wrap the line. And something like "Startb…" would be really bad.

**Comment by violasong** [comment]
Thanks all for this great feedback.

Especially the note on German translations, which does seem like it might be reason enough to avoid the Save to row 😅.

In user testing so far, 3 of 3 participants didn't like option 2 because they felt that the options seemed hidden, or that it felt complicated to have some things hidden. I'm going to try to improve my questions/presentation about this mockup because it might just be that it looks bad on "paper."

**Comment by regs01** [comment]
Grid menu is a mess. It's overloaded and unusable. And thank God desktop Firefox ditched it.

Aside that I think current menu is fine where it is. It doesn't need additional grid style line.

**Comment by kleinph** [comment]
I also prefer the second version (with the save-to options in a sub menu).

Mostly because these save-to options are used very rarely or never by me (with one exception, bookmarks), but are thus in a very prominent position. (Of course this argument is only valid for me). So I would base the decision maybe more on the average usage count of these features than on its discoverability?

Side note: Of course German words can also have line breaks, like:
```
Startbild-
schirm
```
Which also looks not very nice and I think it is not done often in UI contexts. Also translators would have to consider correct hyphenation rules (you can't split a word anywhere).

**Comment by violasong** [comment]
Sounds like the "Save to" row will not work for German and certain other languages. So I'm currently contemplating these two options for a German layout that could also be chosen by other localizers.

<img width="1403" alt="german options" src="https://user-images.githubusercontent.com/55994/105532168-532dd480-5c9f-11eb-8aac-60c2637561fe.png">

**Comment by dglttr** [comment]
Thanks for the update :)
I still much prefer the version on the left, which looks a lot cleaner and more compact.

An idea, because you mentioned that study participants felt the options were hidden: Use an icon that is more associated with saving. The list icon is rather generic and could mean anything if you don't read the string.
Maybe not the standard memory card icon (which is more closely related to saving files IMO), but possibly the "add box" or "move to inbox" icons below are paths to explore?

![grafik](https://user-images.githubusercontent.com/46764284/105534423-c2321a00-5ced-11eb-90b3-0652f99a345c.png)

Another path might be exploring whether these options and the share dialog might be merged?
They are both related to getting the URL somewhere, and some share targets (such as `Save to Pocket`) are actually doing exactly the same thing.
In that case, the icon to be used for the submenu might simply be the share icon, and on pressing the menu, a (beefed up) version of the share sheet opens.
What do you think?

**Comment by klint** [comment]
Still option 2 for me, with a title to the submenu (as any other submenu). 

Option 1 is not extensible, that's a blocker. And I find it far too wide on the screen.

PS: the Report Site Issue entry seems to be gone. For good? :)

**Comment by cadeyrn** [comment]
@violasong 

> So I'm currently contemplating these two options for a German layout that could also be chosen by other localizers.

I don't think localizers shouold "chose" the layout, especially since what fits depends on the device as well. The menues already have a dynamic width. And AFAIK there is a reason why it doesn't use the full width (but I don't remember what the reason is).

**Comment by violasong** [comment]
Re: Save to icon: True, the list icon was non-ideal, though after internal feedback I'm now actually going with no icons for most of the options.

Report Site Issue: I realized this was only in pre-release versions, so I'm removing it from these mockups so I can focus on the main experience.

So, I'm no longer considering a separate layout for German, because after discussing this with my colleagues I realized that the maintenance cost to this was too great. Because of this, and extensibility reasons, I'm leaning toward the submenu variant. However, it still has two issues: Some users in testing still say they would tap the top "Bookmarks" button when adding a bookmark. Also, I know users were uncomfortable with having things hidden in a menu, even if those things weren't something they used.

Here's where I'm at currently, but still iterating. The thought is that maybe it helps to group "Save page to" next to what is basically "Saved stuff."

<img width="778" alt="Screen Shot 2021-01-26 at 4 18 06 PM" src="https://user-images.githubusercontent.com/55994/105923931-14c34d00-5ff2-11eb-8fd9-64a9cb97c540.png">

**Comment by sheikh-azharuddin** [comment]
Hey why not expand the menu horizontally to full width??
This will reduce the overall height and add more items... Like we have in Firefox lite or microsoft edge.. 

![Screenshot_20210127-081806481](https://user-images.githubusercontent.com/58127841/105935580-37984480-6078-11eb-9099-75ea1b4f640b.jpg)

**Comment by kleinph** [comment]
> I know users were uncomfortable with having things hidden in a menu, even if those things weren't something they used.

So I guess there are also users who are uncomfortable with things they don't use cluttering up their menus?

**Comment by regs01** [comment]
> Hey why not expand the menu horizontally to full width??

Overloaded, hard to use one-hand.

**Comment by klint** [comment]
Thanks @violasong 

>  Also, I know users were uncomfortable with having things hidden in a menu, even if those things weren't something they used.
Why not just highlight this "Save to" menu entry with a blue dot until it is used for the first time?


Regarding the order of menu items, I have a question: how are they sorted? are the ones above the least frequently used according to your research (page related actions) and the ones below the most frequently used (general actions)? I'm assuming of course that the ones above are more difficult to reach than the ones below.
Thanks :)

**Comment by cadeyrn** [comment]
> hard to use one-hand.

This is definitely not true. In fact the menu of Firefox Lite is much easier to use one-hand than the current menu of Firefox. Just try it.

@violasong There is still the open question of where the menu items of add-ons will be located. I think it's a fundamental question because it's an issue in all your proposals. The icon row does not look like there would be a submenu on the add-ons item and without any visual indication it wouldn't make much sense to have a submenu for one of these buttons in the icon row but not for the others.

**Comment by regs01** [comment]
> This is definitely not true and to be honest I don't undestand why you think so. In fact the menu of Firefox Lite is much easier to use one-hand than the current menu of Firefox. Just try it.

It's true. For hence reason it's easier to use Chrome and current Firefox menu.

**Comment by cadeyrn** [comment]
> It's true. For hence reason it's easier to use Chrome and current Firefox menu. 

Then you have to explain it. As I already said: I don't understand why you think so. You can use the full width of every row, to a full width menu can't make it more difficult to use than a non full width menu. And since the menu of Firefox Lite has a smaller height it's also easier to reach every menu item, compared to the currrent and much higher menu of Firefox.

As a consequence for me it's not possible at all to reach every menu item with only one hand in Firefox, but in Firefox Lite it's super easy. So how is it possible that you say the opposite? What am i missing?

**Comment by sheikh-azharuddin** [comment]
Also I found nice implementation in Samsung internet browser... Just 4 rows of icons(icons can be rearranged) .. Smaller height and clean UI

![Screenshot_20210129-202026463](https://user-images.githubusercontent.com/58127841/106289772-b0adbc80-626f-11eb-9c55-2b4739e079a4.jpg)

**Comment by violasong** [comment]
Hi all,

After going through several more rounds of UX review with my colleagues, we re-focused on the core issues of this design: improving organization with semantic groupings, improving scannability by simplifying visuals, and solving the confusion around bookmarking. We also discussed a unified experience across desktop and mobile, which will include moving away from our current reliance on icons and reducing submenus where possible. 

That brought this design back to the list layout, with the addition of a clearer bookmarking interface. It may not be as fancy as some of the previous proposals, but we're erring on the side of being straightforward and text-focused. This design went through 8 more user tests and received positive reactions and a 100% bookmarking success rate. Here's the final design:

<img width="2590" alt="menu final" src="https://user-images.githubusercontent.com/55994/106664579-18a72000-655a-11eb-9f9c-2919dfef1606.png">

**Comment by cadeyrn** [comment]
@violasong Thanks for the update.

I think "improving scannability" and "moving away from our current reliance on icons" is really a big contradiction. Most people are "visual people" and icons help a lot to find menu items quickly. Now users have to **read every menu item** to find the correct one. 

Also the proposed bookmarks menu item is very confusing for two reasons:

1. It's a very unusual way for a menu item with two different functions. And to be honest the purple part on the right side doesn't look like it would belong there.

2. The icon and the text have opposite meanings:

a) empty star and "Add": negative state (no bookmark) + positive action (make it a bookmark)
b) filled star and "Remove": positive state (bookmark) + negative action (remove the bookmark)

While the _functionality_ is not different from the current implementation the combination of icon and text makes this a bit confusing.

**Comment by violasong** [comment]
@cadeyrn thanks for your feedback, as always. Re: the bookmark add/remove, I actually realized there's an error here, and it should say Edit instead of remove. I'll think about how I can improve the state issue beyond that.

As for the icons, I understand the concern. I'm expecting it's more that people will scan the first letter, or first few letters, rather than reading every word. I know my colleagues have done a lot of work on this decision - I'll read up on it and see if there's more that I can share.

**Comment by sheikh-azharuddin** [comment]
Increase the list a bit longer for it to go out of the screen🤣🤣... I don't understand how can the test users compare desktop with mobile... Probably they are using 4" device... The bottom address bar was added with the sole purpose of easy accessibility with a single hand for large devices.. Increasing the list simply defeat the purpose.. You operate desktop with a mouse but mobile device is different. So no comparison of mobile with desktop👎🏼

**Comment by Dunexus** [comment]
Hello @violasong ,
> We also discussed a unified experience across desktop and mobile, which will include moving away from our current reliance on icons and reducing submenus where possible.

On desktop, the menu uses icons. It is not clear how removing icons improves scannability and unifies the experience. It looks like it is the exact opposite.

![image](https://user-images.githubusercontent.com/13906248/106722823-b821ec80-6606-11eb-9da1-5525170fe257.png)

**Comment by klint** [comment]
@violasong  
Also, on desktop, "New page" is almost the closest menu option to the hamburger icon that opens the menu. Hence, no big mouse move is needed.
Whereas on mobile, "New tab" is the furthest option from the menu icon. Not very handy to be used with one hand.

**Comment by cadeyrn** [comment]
@Dunexus:

> On desktop, the menu uses icons

I don't know the final mockups but the current implementation of the upcoming Proton menu does not contain any icons so far.

@violasong:

> As for the icons, I understand the concern. I'm expecting it's more that people will scan the first letter, or first few letters, rather than reading every word.

Yes, our brain is very smart and we don't need to read every letter. But it still makes a big difference to have icons or not to have icons. Removing the icons will definitively not improve the scannability. We know from science that for most people images are faster to scan than text becasue images speak directly to the visual center of our brain and do not need to be combined with other signs like text. Scanning text is much more complex for the human brain. So please reconsider this.

**Comment by klint** [comment]
Also, quite sad that the final layout will not make it easy to implement #15450 (Always allow "add to home screen", even if the site can be installed as a PWA).

<OFF-TOPIC or maybe not quite>
As a side note, after many months of Fenix usage, it appears that:
1) I NEVER used collections and prefer to leave all tabs open in tab tray (at least they will be synced that way) ;)  
2) for what it is about here, I'm mad at Fenix every other day because it won't let me create a shortcut to a specific article in my newspaper website (which is available as PWA of course).
</OFF-TOPIC>

**Comment by CharoSW** [comment]
I feel like that horizontal bookmarks/history/downloads icon and text bar is a lot easier to see and discern the function, and also easier to hit with fingers and not have to worry about scrolling the menu.  I also second @cadeyrn 's point about the bookmarks item being weird with two parts.  It's a confusing element that I as a user would have to think to figure out what it's doing, running counter to your goal of making the user think less, and its also a bad hit target.  If I am using a phone one handed with my right hand then I have a decent chance of adding (moderately annoying) or removing (extremely annoying) the page as a bookmark vs opening the bookmarks menu like I would want.

**Comment by sheikh-azharuddin** [comment]
> Yes, our brain is very smart and we don't need to read every letter. But it still makes a big difference to have icons or not to have icons. Removing the icons will definitively not improve the scannability. We know from science that for most people images are faster to scan than text becasue images speak directly to the visual center of our brain and do not need to be combined with other icons like multiple letters has to be combined to a word to make sense. Scanning text is much more complex for the human brain. So please reconsider this.

In simple term a picture speaks thousand word.. So icons are better than text at any time

**Comment by JoGarWeb** [comment]
How many test users do you rely on to make decisions? Less than 10? So many good proposals throughout these long months so that the design ends up practically as loaded and now without icons. 
Is all I will say...

**Comment by violasong** [comment]
Hey all, I understand there's frustration with this outcome after seeing the intermediate mockups that many of you preferred. 

To explain my understanding of the icons decision coming to both desktop and mobile: Our researchers have conducted 9 studies related to this issue in the last few years, and they found that people recognized almost none of the icons without the text. The new direction is to use only highly recognizable icons, in places where they are necessary, and avoid the others, because they add noise for most people.

**Comment by kleinph** [comment]
As others already pointed out, there are a few problems with it.

At first I agree with @cadeyrn that the removal of icons is contradictory to "improving scan-ability". Also keep in mind that this relies on the wording and thus scan-ability might be sensitive to wording changes. There may be also (very few) users who use the browser in different languages (on different devices). (But yes, this is a bit of a extreme corner case.) Icons are universal and not affected by wording or language changes.

The other thing, others already pointed out is the double function of the bookmark button.

The last thing the concerns me are the hidden menu entries. If I understand this correctly, the menu entry "Add to Home Screen" is half visible in its default state. I assume this should make the hidden entries discoverable? But doesn't this look a bit wrong in its default state?

(As I said before, I think [this design](https://github.com/mozilla-mobile/fenix/issues/14320#issuecomment-763102740) was the best one, and I don't really understand what's wrong with it.)

**Comment by sheikh-azharuddin** [comment]
> Hey all, I understand there's frustration with this outcome after seeing the intermediate mockups that many of you preferred.
> 
> To explain my understanding of the icons decision coming to both desktop and mobile: Our researchers have conducted 9 studies related to this issue in the last few years, and they found that people recognized almost none of the icons without the text. The new direction is to use only highly recognizable icons, in places where they are necessary, and avoid the others, because they add noise for most people.

And what's the issue with having icons and text like in previous mockups?? Even the Firefox lite UI looks better

And there is no reason to expand the UI to full width which could possibly reduce the height

**Comment by eliserichards** [comment]
Thanks for your input, y'all! I'm closing this ticket as _done_ since the designs are finalized 😄 👍

**Comment by sheikh-azharuddin** [comment]
Final closing note:
After months of mocks and design Firefox decided to simply extend the menu height by adding few more items making  it more harder to reach top items with a single finger for large devices!

Speaking of similarities not sure why Mozilla added the bottom address bar in fenix when desktop does not have one!

Anyway Mozilla does what they want not what practical and useful..

**Comment by JoGarWeb** [comment]
Final Notes: After months of good designs, fewer than 20 people decided to copy Chrome's design.

Thanks for nothing, I won't give you ideas again. As a suggestion without going to copy the design of that competitor's browser, do the same with the current empty and useless homescreen. 

![Firefox Chromium](https://user-images.githubusercontent.com/7904563/106929910-92feae00-66da-11eb-99ff-2bf2033b5973.png)

**Comment by violasong** [comment]
Folks... it takes time and emotional fortitude to share the full design process, which is why almost no designers do this. I, too, would have loved to ship one of the more collaborative designs which you all helped with. But I'm still working within a company that has strategies and goals which I'm beholden to, and that caused me to make changes. Please don't ruin my day over this or it will be much harder to share the design process next time.

**Comment by sheikh-azharuddin** [comment]
@violasong  I understand that why I said Mozilla does what they want not what is practical and useful.. In short a company which advertise it is build for people and in reality works as per their ideology.. . A fox in sheep skin!
